eventsourcing/src/main/java/org/axonframework/eventsourcing/annotation/AnnotationBasedEventCriteriaResolver.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.annotation;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.common.ReflectionUtils;
+import org.axonframework.common.annotation.AnnotationUtils;
+import org.axonframework.common.infra.ComponentDescriptor;
+import org.axonframework.common.infra.DescribableComponent;
+import org.axonframework.eventsourcing.CriteriaResolver;
+import org.axonframework.eventsourcing.eventstore.EventCriteria;
+import org.axonframework.eventsourcing.eventstore.Tag;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Annotation-based {@link CriteriaResolver} implementation which resolves {@link EventCriteria} based on the given
+ * {@code id} for loading the entity. This is the default when using the {@link EventSourcedEntity} annotation.
+ * <p>
+ * There are various ways to define how the {@link EventCriteria} should be resolved. In order of precedence:
+ * <ol>
+ *     <li>
+ *         By defining a static method in the entity class annotated with {@link EventCriteriaBuilder} which returns an
+ *         {@link EventCriteria} and accepts the {@code id} as a parameter. This method should be static and return an
+ *         {@link EventCriteria}. Multiple methods can be defined with different id types, and the first matching method
+ *         will be used.
+ *     </li>
+ *     <li>
+ *         If no matching {@link EventCriteriaBuilder} is found, the {@link EventSourcedEntity#tagKey()} will be used as the tag key, and the {@link Object#toString()} of the id will be used as value.
+ *     </li>
+ *     <li>
+ *         If the {@link EventSourcedEntity#tagKey()} is empty, the {@link Class#getSimpleName()} of the entity will be used as tag key, and the {@link Object#toString()} of the id will be used as value.
+ *     </li>
+ * </ol>
+ *
```

Comment:
Something that wasn't immediately apparent to me, is that this `AnnotationBasedEventCriteriaResolver` essentially groups several `CriteriaResolver` instances for the given `entityType`. I think it would be good to describe that in the class-level JavaDoc.

Reply:
Adding javadoc